lxd/api_internal.go

```diff
@@ -108,6 +112,49 @@ var internalRAFTSnapshotCmd = APIEndpoint{
 	Get: APIEndpointAction{Handler: internalRAFTSnapshot},
 }
 
+var internalImageRefreshCmd = APIEndpoint{
+	Path: "image-refresh",
+
+	Get: APIEndpointAction{Handler: internalRefreshImage},
+}
+
+var internalImageOptimizeCmd = APIEndpoint{
+	Path: "image-optimize",
+
+	Post: APIEndpointAction{Handler: internalOptimizeImage},
+}
+
+type internalImageOptimizePost struct {
+	Image api.Image `json:"image" yaml:"image"`
+	Pool  string    `json:"pool" yaml:"pool"`
+}
+
+func internalOptimizeImage(d *Daemon, r *http.Request) response.Response {
+	req := &internalImageOptimizePost{}
+
+	// Parse the request.
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	err = imageCreateInPool(d, &req.Image, req.Pool)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.EmptySyncResponse
+}
+
+func internalRefreshImage(d *Daemon, r *http.Request) response.Response {
+	err := autoUpdateImages(context.Background(), d)
```

Comment:
@monstermunchkin shouldnt this be the passed the context from the daemon rather than the base context, in case the daemon itself is shutdown?